### DISCUSSION TITLE
Add Slack support for PublishPress Revisions with interactive messages

### DISCUSSION THREAD

**Discussion by andergmartins** [publishpress/PublishPress-Planner#1287] [Feature Requests]
I would like to suggest adding support to Slack's message actions, which displays buttons after a notification. The notifications module could support Revisionary's events, and trigger Slack notifications with buttons to approve, decline, schedule, etc... right inside Slack.

Related to https://github.com/publishpress/Revisionary-Pro/issues/2

**Comment by stevejburge** [comment]
@andergmartins 100% agreement here. 

If we allow Revisionary to use PublishPress notifications, it becomes much more powerful

**Comment by andergmartins** [comment]
Nice. We could use interactive messages:

https://api.slack.com/interactive-messages

![buttons_simple_example](https://user-images.githubusercontent.com/407131/64022758-44c0a500-cb0d-11e9-890c-f6e41a74063a.gif)

**Comment by andergmartins** [comment]
Just another idea, for adding more interaction we can use the Block Kit on Slack:

https://api.slack.com/block-kit

And have different fields right in the message:

<img width="1011" alt="Screen Shot 2019-08-30 at 09 39 48" src="https://user-images.githubusercontent.com/407131/64021476-72f0b580-cb0a-11e9-805e-96361a56af41.png">

**Comment by agapetry** [comment]
@andergmartins  I agree with this direction. Toward actual implementation, here are some factors to consider.  Currently, PublishPress Revisions supports the following notification behavior:

- Plugin Settings specify which roles receive notification.
- Plugin Settings specify whether revision submitter can skip notifications per-user.
- If PressPermit Pro is active, skip recipients who cannot fully edit the post being revised.
- If PressPermit Pro is activated and [Pending Revision Monitors] group populated with at least one user, use that member list instead of configured roles. Actual recipients are still filtered by editing capability on revised post.

That's the email recipient logic for notification of a Pending Revision submission.   There is one other notification which supports an equivalent PressPermit Pro user group (Scheduled Revision publication) and one which does not (Pending Revision Approval).  We should probably add notifications for revision scheduling and, looking ahead, Rejection or Needs Work. 

How do we foresee an integration with PublishPress Notifications supporting this behavior?  Would you prefer to let PPR continue to handle the recipient customization UI in the post editor, or make PPN handle that too?  If the latter, we would need to synchronize UI to avoid a jarring visual change when someone moves from standalone notifications to PPN integration.

**Comment by stevejburge** [comment]
@agapetry IMHO, we'd take PressPermit out of this, and perhaps even Revisions ... we'd just use PublishPress Notifications

**Comment by agapetry** [comment]
I guess @andergmartins and I will just need to discuss the logistics of doing that without losing the current functionality.  Let me boil down what I said in that last comment.  In the current implementation for Pending Revision submission:

1. The configured list of notification recipients is filtered down to those who would be able to publish the revision.  This supports simple configuration while preventing irrelevant notifications.

2. When a post is edited for pending revision submission, that filtered recipient list is displayed within the post editor as a preview of who will be notified.  The revision submitter can remove recipients before submitting the revision (unless that ability is disabled via plugin settings).

**Comment by stevejburge** [comment]
@andergmartins @agapetry This sounds like a 2020 issue to me :)
